name: Trace Blame CI

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  flake_and_test:

    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']

    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip
          pip install -r requirements.txt
          pip install pytest
          pip install flake8
      - name: Lint with flake8
        run: |
          flake8 --exclude .git,__pycache__,venv --extend-ignore E501
      - name: Test with pytest
        run: |
          git status
          git log --oneline
          git blame tests/tests.py -L 18,18
          git blame HEAD tests/tests.py -L 18,18
          pytest tests/tests.py -vv -s
